### PR TITLE
Supporting ecosystem.config.js. This allows to use javascript and module.exp…

### PR DESCRIPTION
**Pull request by igorushko**
| Q             | A
| ------------- | ---
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #2433
| License       | MIT
| Doc PR        | https://github.com/pm2-hive/pm2-hive.github.io/pulls

…orts in config file to avoid copy-paste and etc

### CODE REVIEW

File: lib/Common.js [hunk 1]
```diff
@@ -199,6 +199,8 @@ Common.isConfigFile = function(filename) {
     return 'json';
   if (filename.indexOf('.yml') > -1 || filename.indexOf('.yaml') > -1)
     return 'yaml';
+  if (filename.indexOf('.js') != -1)
+	return 'js';
```
Comment (soyuka) [inline lib/Common.js]: indent

File: lib/Common.js [hunk 2]
```diff
@@ -230,7 +232,11 @@ Common.parseConfig = function(confObj, filename) {
   else if (filename.indexOf('.yml') > -1 ||
            filename.indexOf('.yaml') > -1) {
     return yamljs.parse(confObj.toString());
+  }else if (filename.indexOf('.js') > -1){
```
Comment (soyuka) [inline lib/Common.js]: missing space after {

File: lib/Common.js [hunk 3]
```diff
@@ -230,7 +232,11 @@ Common.parseConfig = function(confObj, filename) {
...
+  	delete require.cache[require.resolve(filename)];
+  	return require(filename);
```
Comment (soyuka) [inline lib/Common.js]: indent

File: lib/Common.js [hunk 4]
```diff
@@ -199,6 +199,8 @@ Common.isConfigFile = function(filename) {
     return 'json';
   if (filename.indexOf('.yml') > -1 || filename.indexOf('.yaml') > -1)
     return 'yaml';
+  if (filename.indexOf('.config.js') != -1)
+	return 'js';
```
Comment (vmarchaud) [inline lib/Common.js]: need indent here

Comment (vmarchaud) [inline lib/Common.js]: need indent here.

**Comment by vmarchaud** [comment]
Seems good to me, waiting for test.

**Comment by Unitech** [comment]
nice! we should update http://pm2.keymetrics.io/docs/usage/application-declaration/

**Comment by Unitech** [comment]
landed in main + doc updated

pm2@2.1.4 (main):

```bash
$ npm install pm2 -g
$ pm2 update
```

**Comment by doganyazar** [comment]
As I understand from the code the filename has to have ".config.js" as extension so something like "app.js" does not work. Is there a reason to force ".config" part? I have been debugging my app last couple of hours to find this out. At least it would be nice to mention this in the documentation that .config part is necessary.

**Comment by vmarchaud** [comment]
@doganyazar Because pm2 need to know when your entry file is a ecosystem file or just a javascript file. The `.config.js` part is specified in the documentation right [there](http://pm2.keymetrics.io/docs/usage/application-declaration/#javascript-format)

**Comment by doganyazar** [comment]
Thanks for the answer @vmarchaud. The confusing part for me was that documentation does not say .config part is a must but yeah maybe it is trivial for the others :)

**Comment by vmarchaud** [comment]
@doganyazar Its written in the documentation that i linked

**Comment by doganyazar** [comment]
My bad! Sorry for wasting your time.

**Comment by vmarchaud** [comment]
@doganyazar No problem :)

**Pull request merged**